dot/state/grandpa.go

```diff
@@ -233,13 +233,11 @@ func (s *GrandpaState) SetNextPause(number *big.Int) error {
 	return s.db.Put(pauseKey, number.Bytes())
 }
 
-// GetNextPause returns the block number of the next grandpa pause, nil if there is no upcoming pause
+// GetNextPause returns the block number of the next grandpa pause.
+// It returns an error on failure, which is contains the
+// chaindb.ErrKeyNotFound error if the key is not found.
 func (s *GrandpaState) GetNextPause() (*big.Int, error) {
 	num, err := s.db.Get(pauseKey)
-	if err == chaindb.ErrKeyNotFound {
-		return nil, nil
-	}
-
```

Comment:
we shouldn't remove this, we should ignore this error

Reply:
I changed the callers to check for `chaindb.ErrKeyNotFound` and adapt their behavior.
Should I still revert this and nolint this linter?

Reply:
If you check the callers of this function, when they get a `nil` error and a `nil` worker, it continues in their for loops.
So techically, it's true this is not really an error in the eyes of the current callers, although it could be in a future eventual caller :thinking:

Reply:
@kishansagathiya we can ignore this because if the error is `KeyNotFound`, then there is no grandpa authorities pause coming up, so we don't need to do anything.

@qdm12  ok, as long as the logic is the same, that should be fine then

Reply:
The fact @kishansagathiya asked the question of why we ignore the error is also a good point we shouldn't discard error and return i.e. `nil, nil` since it can cause confusion.

Alternatively, instead of returning the not found error, we can change the signature from

```go
GetNextPause() (blockNumber *big.Int, err error)
```

to

```go
GetNextPause() (blockNumber *big.Int, exists bool, err error)
```

? :thinking:

Reply:
I would rather just ignore the error

Reply:
can we just add a comment saying `if the key is not found in the database, that's fine as it means there is no upcoming grandpa pause`

Reply:
Actually `GetNextResume` and `GetNextPause` are both only used in the test `TestHandler_GrandpaPauseAndResume`. Hence the caller code wasn't changed, my bad. Anyway, I reverted it to return `nil, nil` for both and added a not nil check in the test.